### ISSUE TITLE
Presence of version 0.18.1 of azure_mgmt_network gem causes provisioning to fail

### ISSUE THREAD

**Issue by SathyaBhat** [test-kitchen/kitchen-azurerm#99]
We use test kitchen with the Kitchen AzureRM driver to provision an instance in Azure with a public IP. Noticed that with version 0.18.1 of azure_mgmt_network gem, the kitchen provisioning fails. Below are the logs

```
-----> Cleaning up any prior instances of <datadog-azure-check-enabled-azure-centos>
-----> Destroying <datadog-azure-check-enabled-azure-centos>...
       Finished destroying <datadog-azure-check-enabled-azure-centos> (0m0.00s).
-----> Testing <datadog-azure-check-enabled-azure-centos>
-----> Creating <datadog-azure-check-enabled-azure-centos>...
/apps/home/jenkins/.azure/credentials was not found or not accessible.
       Creating Resource Group: kitchen-datadog-azure-check-enabled-azure-centos-20181129T105046
       Creating deployment: deploy-9ebeebe854961353
       Using custom vnet: <snip>
       Adding public key from <snip>
       Resource Microsoft.Resources/deployments 'pid-18d63047-6cdf-4f34-beed-62f01fc73fc2' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'test-kitchen' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'test-kitchen' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'test-kitchen' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'test-kitchen' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'test-kitchen' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'test-kitchen' provisioning status is Running
       Resource Microsoft.Compute/virtualMachines 'test-kitchen' provisioning status is Running
       Resource Template deployment reached end state of 'Succeeded'.
rake aborted!
Kitchen::ActionFailed: Failed to complete #create action: [undefined method `get' for #<Azure::Network::Mgmt::V2017_03_30::PublicIPAddresses:0x0000000447e510>
Did you mean?  gets
               gem]
/apps/home/jenkins/gems/kitchen-azurerm-0.14.4/lib/kitchen/driver/azurerm.rb:256:in `create'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:484:in `public_send'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:484:in `block in perform_action'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:551:in `synchronize_or_call'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:513:in `block in action'
/usr/lib/ruby/2.3.0/benchmark.rb:293:in `measure'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:512:in `action'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:484:in `perform_action'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:394:in `create_action'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:382:in `block (2 levels) in transition_to'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/lifecycle_hooks.rb:45:in `run_with_hooks'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:381:in `block in transition_to'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:380:in `each'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:380:in `transition_to'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:162:in `verify'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:191:in `block in test'
/usr/lib/ruby/2.3.0/benchmark.rb:293:in `measure'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:187:in `test'
cookbook-datadog/Rakefile:53:in `block (3 levels) in <top (required)>'
/usr/lib/ruby/2.3.0/delegate.rb:83:in `each'
/usr/lib/ruby/2.3.0/delegate.rb:83:in `method_missing'
cookbook-datadog/Rakefile:51:in `block (2 levels) in <top (required)>'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:271:in `block in execute'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:271:in `each'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:271:in `execute'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:213:in `block in invoke_with_call_chain'
/usr/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:193:in `invoke_with_call_chain'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:247:in `block (2 levels) in invoke_prerequisites_concurrently'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/promise.rb:64:in `chore'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/promise.rb:34:in `block in value'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/promise.rb:32:in `synchronize'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/promise.rb:32:in `value'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:250:in `each'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:250:in `invoke_prerequisites_concurrently'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:233:in `invoke_prerequisites'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:212:in `block in invoke_with_call_chain'
/usr/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:193:in `invoke_with_call_chain'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:182:in `invoke'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:160:in `invoke_task'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:116:in `each'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:116:in `block in top_level'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:125:in `run_with_threads'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:110:in `top_level'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:83:in `block in run'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:186:in `standard_exception_handling'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:80:in `run'
/apps/home/jenkins/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'
/usr/local/bin/rake:23:in `load'
/usr/local/bin/rake:23:in `<main>'

Caused by:
NoMethodError: undefined method `get' for #<Azure::Network::Mgmt::V2017_03_30::PublicIPAddresses:0x0000000447e510>
Did you mean?  gets
               gem
/apps/home/jenkins/gems/kitchen-azurerm-0.14.4/lib/kitchen/driver/azurerm.rb:256:in `create'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:484:in `public_send'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:484:in `block in perform_action'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:551:in `synchronize_or_call'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:513:in `block in action'
/usr/lib/ruby/2.3.0/benchmark.rb:293:in `measure'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:512:in `action'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:484:in `perform_action'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:394:in `create_action'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:382:in `block (2 levels) in transition_to'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/lifecycle_hooks.rb:45:in `run_with_hooks'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:381:in `block in transition_to'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:380:in `each'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:380:in `transition_to'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:162:in `verify'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:191:in `block in test'
/usr/lib/ruby/2.3.0/benchmark.rb:293:in `measure'
/apps/home/jenkins/gems/test-kitchen-1.23.2/lib/kitchen/instance.rb:187:in `test'
cookbook-datadog/Rakefile:53:in `block (3 levels) in <top (required)>'
/usr/lib/ruby/2.3.0/delegate.rb:83:in `each'
/usr/lib/ruby/2.3.0/delegate.rb:83:in `method_missing'
cookbook-datadog/Rakefile:51:in `block (2 levels) in <top (required)>'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:271:in `block in execute'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:271:in `each'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:271:in `execute'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:213:in `block in invoke_with_call_chain'
/usr/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:193:in `invoke_with_call_chain'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:247:in `block (2 levels) in invoke_prerequisites_concurrently'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/promise.rb:64:in `chore'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/promise.rb:34:in `block in value'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/promise.rb:32:in `synchronize'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/promise.rb:32:in `value'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:250:in `each'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:250:in `invoke_prerequisites_concurrently'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:233:in `invoke_prerequisites'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:212:in `block in invoke_with_call_chain'
/usr/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:193:in `invoke_with_call_chain'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/task.rb:182:in `invoke'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:160:in `invoke_task'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:116:in `each'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:116:in `block in top_level'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:125:in `run_with_threads'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:110:in `top_level'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:83:in `block in run'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:186:in `standard_exception_handling'
/apps/home/jenkins/gems/rake-12.3.1/lib/rake/application.rb:80:in `run'
/apps/home/jenkins/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'
/usr/local/bin/rake:23:in `load'
/usr/local/bin/rake:23:in `<main>'
Tasks: TOP => azure => azure:datadog```

Rolling back to 0.17.0 of the gem resulted in the provisioning working as usual

**Comment by stuartpreston** [comment]
Thanks for the detailed analysis, I’ll try and figure out what we need to do to support the updated azure_mgmt gems, or pin the version at 0.17.0

**Comment by stuartpreston** [comment]
Issue being tracked at https://github.com/Azure/azure-sdk-for-ruby/issues/1968
We will pin to 0.18.0 for now

**Comment by stuartpreston** [comment]
This has been released to Rubygems in kitchen-azurerm 0.14.6
